llvm/lib/Transforms/Scalar/LoopLoadElimination.cpp

```diff
@@ -448,6 +448,10 @@ class LoadEliminationForLoop {
         new LoadInst(Cand.Load->getType(), InitialPtr, "load_initial",
                      /* isVolatile */ false, Cand.Load->getAlign(),
                      PH->getTerminator()->getIterator());
+    // We don't give any debug location to Initial, because it is inserted
+    // into the loop's preheader. A debug location inside the loop will cause
+    // a misleading stepping when debugging. The test preserving-debugloc-store
```

Comment:
Hmm, is this an existing test somewhere? Or maybe it's meant to be attached to this PR...? I don't see it in the file list currently.

Reply:
Oh, I made a mistake here :(. I forgot to make the test name consistent after I renamed the attached test (I changed the name from "preserving-" to "update-", because there is a drop for the `Initial`). I'll fix it.